### PR TITLE
[WFLY-18463] ee-security Quickstart Common Enhancements CY2023Q3

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
Issue: https://issues.redhat.com/browse/WFLY-18463

### CODE REVIEW

File: ee-security/pom.xml [hunk 1]
```diff
@@ -29,6 +29,16 @@
         <relativePath/>
     </parent>
 
+    <properties>
```
Comment (emmartins) [inline ee-security/pom.xml:32]: This should follow https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.q4f7r5l2mqax

```
<properties>
      <!-- the version for the Server -->
      <version.server>30.0.0.Beta1</version.server>
      <!-- The versions for BOMs, Packs and Plugins -->
      <version.bom.ee>${version.server}</version.bom.ee>
      <version.pack.cloud>5.0.0.Beta1</version.pack.cloud>
      <version.plugin.wildfly>4.2.0.Final</version.plugin.wildfly>
  </properties>

```

File: ee-security/pom.xml [hunk 2]
```diff
@@ -285,6 +264,30 @@
...
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
         <profile>
```
Comment (emmartins) [inline ee-security/pom.xml:291]: please remote this profile, no longer used or needed

File: ee-security/pom.xml [hunk 3]
```diff
@@ -250,29 +239,19 @@
...
-                                    <location>org.wildfly.cloud:wildfly-cloud-galleon-pack:${version.cloud.fp}</location>
+                                <feature-pack>                                    
+                                    <location>org.wildfly.cloud:wildfly-cloud-galleon-pack:${version.pack.cloud}</location>
                                 </feature-pack>
                             </feature-packs>
                             <layers>
                                 <layer>cloud-server</layer>
                             </layers>
                             <filename>ROOT.war</filename>
-                            <packaging-scripts>
```
Comment (emmartins) [inline ee-security/pom.xml:267]: Think you still need this for this QS, it reconfigures elytron for the provisioned server

**Review by emmartins — CHANGES_REQUESTED** [review]
Besides changes pointed in PR please change also the following:
* https://github.com/wildfly/quickstart/pull/726/files#diff-32baa650eabeec620b94446ffb8a3cdda73a15963d23329f0266da59983da552L115 this should use property named "version.bom.ee"
* https://github.com/wildfly/quickstart/pull/726/files#diff-32baa650eabeec620b94446ffb8a3cdda73a15963d23329f0266da59983da552L219 failsafe plugin config in provisioned-server no longer used or needed

File: ee-security/pom.xml (hunk 2 above)
Comment (PrarthonaPaul) [reply ee-security/pom.xml:291]: Hi @emmartins 
Thank you for the comments on the PR.
For this comment, are you referring to the arq-remote or the integration-testing profile?

File: ee-security/pom.xml (hunk 1 above)
Comment (PrarthonaPaul) [reply ee-security/pom.xml:32]: Fixed!

File: ee-security/pom.xml (hunk 3 above)
Comment (PrarthonaPaul) [reply ee-security/pom.xml]: Fixed!

File: .github/workflows/ee-security_ci.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    paths:
+      - 'ee-security/**'
+      - .github/workflows/quickstart_ci.yml
+
+jobs:
+  call-quickstart_ci:
+    uses: ./.github/workflows/quickstart_ci.yml
+    with:
+      QUICKSTART_PATH: ee-security
+      MICROPROFILE: false
```
Comment (emmartins) [inline .github/workflows/ee-security_ci.yml:15]: now it should instead be 
TEST_PROVISIONED_SERVER: true

File: ee-security/pom.xml [hunk 5]
```diff
@@ -29,6 +29,15 @@
         <relativePath/>
     </parent>
 
+    <properties>
+        <!-- Version for the server -->
+        <version.server>30.0.0.Beta1</version.server>
+        <!-- The versions for BOMs, Packs and Plugins -->
+        <version.bom.ee>${version.server}</version.bom.ee>
+        <version.pack.cloud>5.0.0.Beta1</version.pack.cloud>
```
Comment (emmartins) [inline ee-security/pom.xml:37]: 5.0.0.Final

**Review by emmartins — CHANGES_REQUESTED** [review]

File: .github/workflows/ee-security_ci.yml (hunk 4 above)
Comment (PrarthonaPaul) [reply .github/workflows/ee-security_ci.yml]: Fixed!

File: ee-security/pom.xml [hunk 6]
```diff
@@ -29,6 +29,15 @@
         <relativePath/>
     </parent>
 
+    <properties>
+        <!-- Version for the server -->
+        <version.server>30.0.0.Beta1</version.server>
```
[the remark this answers is not in the document]
Comment (PrarthonaPaul) [reply ee-security/pom.xml]: Fixed!

File: ee-security/pom.xml (hunk 1 above)
Comment (PrarthonaPaul) [reply ee-security/pom.xml]: Fixed! 
Thank you for your review!

**Review by emmartins — APPROVED** [review]
LGTM, I will merge if all good with CI testing

File: ee-security/pom.xml [hunk 7]
```diff
@@ -250,14 +218,11 @@
                     <plugin>
                         <groupId>org.wildfly.plugins</groupId>
                         <artifactId>wildfly-maven-plugin</artifactId>
-                        <version>${version.wildfly.maven.plugin}</version>
                         <configuration>
                             <feature-packs>
-                                <feature-pack>
-                                    <location>org.wildfly:wildfly-galleon-pack:${version.server}</location>
+                                <feature-pack>                                  <location>org.wildfly:wildfly-galleon-pack:${version.server}</location>
```
Comment (fjuma) [inline ee-security/pom.xml:223]: Just a minor comment, but looks like the formatting is off here and below.

Comment (PrarthonaPaul) [reply ee-security/pom.xml]: Fixed!

**Comment by kabir** [comment]
@emmartins @PrarthonaPaul This PR will not trigger OpenShift CI. The 'ee-security_ci.yml' needs to be prefixed with 'quickstart_' (i.e. quickstart_ee-security_ci.yml) to be picked up

**Comment by PrarthonaPaul** [comment]
> @emmartins @PrarthonaPaul This PR will not trigger OpenShift CI. The 'ee-security_ci.yml' needs to be prefixed with 'quickstart_' (i.e. quickstart_ee-security_ci.yml) to be picked up.
> 
> Although I see now that is a bit fiddly, at least naming it properly will maintain the alphabetical ordering of the files :-)

Sorry, I missed that. 
I can send another PR to fix that. 
Thank you for catching it!

**Comment by emmartins** [comment]
@PrarthonaPaul it's fixed already, no worries

**Pull request merged**